src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -316,12 +318,7 @@ protected function instantiateObject(array &$data, $class, array &$context, \Ref
             $format = null;
         }
 
-        if (
-            isset($context[static::OBJECT_TO_POPULATE]) &&
-            is_object($context[static::OBJECT_TO_POPULATE]) &&
-            $context[static::OBJECT_TO_POPULATE] instanceof $class
-        ) {
-            $object = $context[static::OBJECT_TO_POPULATE];
+        if (null !== $object = $this->extractObjectToPopulate($class, $context, static::OBJECT_TO_POPULATE)) {
             unset($context[static::OBJECT_TO_POPULATE]);
```

Comment:
Would involve passing context by reference into the trait. That's the only reason I didn't do it.